org.adempiere.base/src/org/adempiere/base/BaseActivator.java

```diff
@@ -61,6 +143,9 @@ public void start(BundleContext context) throws Exception {
 		context.registerService(CommandProvider.class.getName(), new StackTraceCommand(), null);
 		
 		blacklistService = new ComponentBlackListService(context);
+		
+		int trackStates = Bundle.RESOLVED | Bundle.STOPPING | Bundle.ACTIVE | Bundle.STARTING;
+		new PoiBundleTracker(context, trackStates).open();
```

Comment:
The example at https://github.com/apache/poi/blob/trunk/osgi/src/main/java/org/apache/poi/osgi/Activator.java just add the provider in activator, why do we need to use bundle tracker here ?

Reply:
example for case merger POI and POI-OOXML into a single bundle

In Idempiere, I need the Bundle Tracker to ensure that both the wrapper bundle already resolved

Reply:
but that's not how it is configure in iDempiere core.

Reply:
Unless there's compelling reason for doing so, I would prefer the simple approach in the link example above and don't have to depends on hard coded bundle name check as well.

Reply:
I'd like a straightforward approach, similar to sample, but need to merged JARs for POI and POI-OOXML for deployment on the idempiere/binary.file

Reply:
If a plugin need a different version of POI, that should be managed in that plugin/bundle, it is not the core job to do that.